signer-wasm/src/ledger.rs

```diff
@@ -224,32 +223,19 @@ pub async fn device_info(transport_wrapper: TransportWrapper) -> Promise {
     let d_result = app.get_device_info().await;
 
     match d_result {
-        Ok(d) => {
-            let answer = JsValue::from_serde(&d)
-                .map_err(|_e| {
-                    Promise::reject(&js_sys::Error::new(
-                        "Error converting answer message to javascript value.",
-                    ))
-                })
-                .unwrap();
-
-            Promise::resolve(&answer)
-        }
+        Ok(d) => Promise::resolve(&ok_or_ret_promise!(
+            JsValue::from_serde(&d),
+            "Error converting error message to javascript value."
+        )),
```

Comment:
IMHO, this macro makes sense. When an error occurs a Promise must be given anyway

Reply:
yes, this is definitely a good improvement